gobblin-core/src/main/java/gobblin/source/extractor/watermark/TimestampWatermark.java

```diff
@@ -37,7 +37,8 @@
   private static final Logger LOG = LoggerFactory.getLogger(TimestampWatermark.class);
   // default water mark format(input format) example: 20140301050505
   private static final String INPUTFORMAT = "yyyyMMddHHmmss";
-  private static final SimpleDateFormat INPUTFORMATPARSER = new SimpleDateFormat(INPUTFORMAT);
+  private final SimpleDateFormat _inputFormatParser;
```

Comment:
Per the coding styles ([Google style guide](https://google.github.io/styleguide/javaguide.html) or sun java guide) in Gobblin,  names should not begin with '_'. In your case, a name `inputFormatParser` is good enough.